packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -161,6 +192,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         };
         updateFocusContextKeys();
         this.shell.activeChanged.connect(updateFocusContextKeys);
+        this.workspaceCommandContribution.onDidCreateNewFile(async uri => this.onDidCreateNewResource(uri));
+        this.workspaceCommandContribution.onDidCreateNewFolder(async uri => this.onDidCreateNewResource(uri));
+    }
+
+    private async onDidCreateNewResource(uri: URI): Promise<void> {
+        const navigator = await this.tryGetWidget();
+        if (!navigator || !navigator.isVisible) {
+            return;
+        }
+        const model: FileNavigatorModel = navigator.model;
+        const parent = await model.revealFile(uri.parent);
+        if (DirNode.is(parent)) {
+            await model.refresh(parent);
+        }
+        const node = await model.revealFile(uri);
+        if (SelectableTreeNode.is(node)) {
+            model.selectNode(node);
```

Comment:
I think we should transfer focus to the navigator here as well. Otherwise one cannot use keyboard to navigate afterwards.
```ts
this.openView({ activate: true });
```

Reply:
done